lib/collection/src/shards/local_shard/clock_map.rs

```diff
@@ -35,8 +37,16 @@ impl ClockMap {
         Ok(clock_map)
     }
 
-    pub fn store(&self, path: &Path) -> Result<()> {
+    pub fn store(&mut self, path: &Path) -> Result<()> {
         file_operations::atomic_save_json(path, self)?;
+        self.changed = false;
+        Ok(())
+    }
+
+    pub fn store_if_changed(&mut self, path: &Path) -> Result<()> {
```

Comment:
Doing extra profiling, I can see that building the path with `oldest_clock_path` shows up although we do not use it.

Could we build the path only if required?

Reply:
I don't think it worth it...